dali/operators/generic/cast.cu

```diff
@@ -29,96 +28,46 @@ namespace dali {
 
 class CastGPU : public Cast<GPUBackend> {
  public:
-  explicit CastGPU(const OpSpec &spec) : Cast<GPUBackend>{spec}, block_setup_{block_volume_scale} {}
-
-  bool SetupImpl(std::vector<OutputDesc> &output_desc, const Workspace &ws) override;
+  explicit CastGPU(const OpSpec &spec) : Cast<GPUBackend>{spec} {}
   void RunImpl(Workspace &ws) override;
-
   ~CastGPU() override = default;
 
- protected:
-  void PrepareBlocks(const Workspace &ws);
-
-  static const int block_volume_scale = 4;
-
  private:
-  using GpuBlockSetup = kernels::BlockSetup<1, -1>;
-  TensorListShape<1> collapsed_shape_;
-
-  GpuBlockSetup block_setup_;
-  std::vector<kernels::CastSampleDesc> samples_;
-
+  static constexpr int kBlockSize = 1024;
+  static constexpr int kLogicalBlockSize = 1024;
```

Comment:
Is it really the best thing? This looks rather suspicious - we run a binary search to copy one element per thread! This seems inefficient.

Reply:
Not tuned yet. I'll update in a bit